app/src/main/java/com/humara/nagar/ui/residents/ResidentsFragment.kt

```diff
@@ -23,23 +40,128 @@ class ResidentsFragment : BaseFragment() {
+                    }
+                    //Check the current item and save the value
+                    chip.isCloseIconVisible = true
+                    previousFilterId = cid
+                } else {
+                    //Nothing is checked
+                }
+            } else {
+                //Enabled chip was disabled
+                group.findViewById<Chip>(previousFilterId).isCloseIconVisible = false
+                previousFilterId = -1
+            }
+        }
+    }
+
+    private fun handleSearchFilter() {
+        binding.inputSearch.addTextChangedListener(object : TextWatcher {
+            override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {}
+
+            override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {}
+
+            override fun afterTextChanged(p0: Editable?) {
+                val residentList = residentsViewModel.residentsResponse.value?.Residents
+                if (p0.toString().isNotEmpty()) {
+                    filter(Utils.toStringWithoutSpaces(p0.toString().lowercase()), residentList)
+                } else {
+                    if (residentList != null) {
+                        residentAdapter.setData(residentList)
+                    }
+                }
+            }
+        })
+    }
+
+    private fun filter(search: String, residentList: ArrayList<Residents>?) {
+        val filterList = mutableListOf<Residents>()
+        if (search.isNotEmpty()) {
+            if (residentList != null) {
+                for (current in residentList) {
+                    if (Utils.toStringWithoutSpaces(current.name!!.lowercase()).contains(search)) {
+                        filterList.add(current)
+                    }
+                }
+            }
+            residentAdapter.setData(filterList)
+        } else
+            residentList?.let { residentAdapter.setData(it) }
+    }
+
+    private fun initView() {
+        binding.apply {
+            recyclerView = residentRCV
+            residentAdapter = AllResidentsAdapter(requireContext()) {
+                //Do something here onClick Card (it -> Residents)
+            }
+            recyclerView.apply {
+                layoutManager = GridLayoutManager(requireContext(), 2)
+                setHasFixedSize(false)
+                adapter = residentAdapter
+            }
+        }
+    }
+
+    private fun initObservers() {
+        residentsViewModel.run {
+            observeProgress(this, false)
+            observerException(this)
+
+            fetchAllFilters()
+            fetchAllResidents()
+
+            residentsResponse.observe(viewLifecycleOwner) {
+                residentAdapter.setData(it.Residents)
+            }
+            filtersResponse.observe(viewLifecycleOwner) {
+                handleChipFilter(it)
+            }
+            residentErrorLiveData.observe(viewLifecycleOwner) {
+                Logger.debugLog("Resident Response API Error: ${it.message}")
```

Comment:
use showErrorDialog function from Base class here in case of api error

Reply:
Resolved.